src/wallet/wallet.go

```diff
@@ -96,40 +119,181 @@ func NewWallet(wltName string, opts Options) (*Wallet, error) {
 		},
 	}
 
+	// Generate addresses
+	if _, err := w.GenerateAddresses(opts.AddressNum); err != nil {
+		return nil, fmt.Errorf("generate addresses failed when creating wallets: %v", err)
+	}
+
+	if !opts.Encrypt {
+		return w, nil
+	}
+
+	if opts.Password == nil {
+		return nil, errors.New("password is required for creating wallet with encryption")
+	}
+
+	if err := w.lock(opts.Password); err != nil {
+		return nil, fmt.Errorf("lock wallet failed: %v", err)
+	}
+
+	// Update the encrypted meta field
+	w.setEncrypted(true)
+
 	return w, nil
 }
 
-// Load loads wallet from given file
-func Load(wltFile string) (*Wallet, error) {
-	w := Wallet{}
-	if err := w.Load(wltFile); err != nil {
-		return nil, err
+// lock encrypts the wallet with password
+func (w *Wallet) lock(password []byte) error {
+	if password == nil {
+		return ErrRequirePassword
+	}
+
+	if w.IsEncrypted() {
+		return ErrWalletEncrypted
 	}
 
-	return &w, nil
+	// Encrypt the seed
+	ss, err := Encrypt([]byte(w.seed()), password)
+	if err != nil {
+		return err
+	}
+
+	w.setSeed(ss)
+
+	// Encrypt the last seed
+	sls, err := Encrypt([]byte(w.lastSeed()), password)
+	if err != nil {
+		return err
+	}
+
+	w.setLastSeed(sls)
+
+	// encrypt private keys in entries
+	for i, e := range w.Entries {
+		se, err := Encrypt(e.Secret[:], password)
+		if err != nil {
+			return err
+		}
+
+		// Set the encrypted seckey value
+		w.Entries[i].EncryptedSeckey = se
+		// Clear the entry.Secret
+		w.Entries[i].Secret = cipher.SecKey{}
```

Comment:
Also do

`for i := Secret { Secret[i] = 0}`to wipe memory before overwriting with empty seckey